src/arch/riscv64/CodeGen.zig

```diff
@@ -172,6 +162,14 @@ const MCValue = union(enum) {
         };
     }
 
+    fn isRegister(mcv: MCValue) bool {
+        return switch (mcv) {
+            .register => true,
+            .register_offset => |reg_off| return reg_off.off == 0,
```

Comment:
So if an offset in `.register_offset` is non-zero, it makes this MCV suddenly non-register? What is the reasoning behind this?

Reply:
The MCV can no longer be acted on as a "value in a register" and needs to be either shifted or manipulated in some way to reach the value we want.

Reply:
Ah right, I see. Naming of the helper function left me a little confused.